internal/pkg/config/config.go

```diff
@@ -137,7 +161,39 @@ func (vacuousMatcher) MatchString(string) bool {
 	return true
 }
 
-// Returns the first non-nil matcher.  If all are nil, returns a vacuousMatcher.
+type fieldTagMatcher struct {
+	Key string
+	Val string
+}
+
+type rawFieldTagMatcher struct {
+	Key string
+	Val string
+}
+
+func (ft *fieldTagMatcher) UnmarshalJSON(bytes []byte) error {
+	// Unknown config fields are not allowed.
+	rawMap := make(map[string]interface{})
+	if err := json.Unmarshal(bytes, &rawMap); err != nil {
+		return err
+	}
+	for label := range rawMap {
+		if !validFieldTagMatcherFields[label] {
+			return fmt.Errorf("%v is not a valid config field", label)
```

Comment:
I think it would be good for the error message to be a bit more verbose, e.g. to include the fact that you were trying to parse field tag configuration when you encountered the invalid field. Also consider listing the valid fields, e.g. `valid fields include: ...`.

Reply:
Added